From 1f91f55f124597dbd9dc47a2ac1ff0cb9b80e2b7 Mon Sep 17 00:00:00 2001
From: Mark Harfouche <mark.harfouche@gmail.com>
Date: Sun, 30 Mar 2025 12:19:47 -0400
Subject: [PATCH] Make the "Help Spyder" Not depend on QtWebEngine

I understand that there are fancy solution and maybe larger
refactors but:

1. We want users to have compatibility in their installations
2. We want to ensure that users have a donation button

I think this hits both

xref: https://github.com/spyder-ide/spyder/pull/23703
---
 spyder/plugins/application/widgets/status.py | 21 ++++++++++++++------
 1 file changed, 15 insertions(+), 6 deletions(-)

diff --git a/spyder/plugins/application/widgets/status.py b/spyder/plugins/application/widgets/status.py
index 80c89caf0e8..66f21c346d6 100644
--- a/spyder/plugins/application/widgets/status.py
+++ b/spyder/plugins/application/widgets/status.py
@@ -23,7 +23,6 @@
 from spyder.utils.icon_manager import ima
 from spyder.utils.qthelpers import start_file
 from spyder.utils.stylesheet import WIN
-from spyder.widgets.browser import WebView
 
 
 class InAppAppealDialog(QDialog, SpyderFontsMixin):
@@ -52,6 +51,7 @@ def __init__(self, parent=None):
             "index.html",
         )
 
+        from spyder.widgets.browser import WebView
         # Create webview to render the appeal message
         webview = WebView(self, handle_links=True)
 
@@ -116,11 +116,20 @@ def _on_click(self):
     # ---- Public API
     # -------------------------------------------------------------------------
     def show_appeal(self):
-        if self._appeal_dialog is None:
-            self._appeal_dialog = InAppAppealDialog(self)
-
-        if not self._appeal_dialog.isVisible():
-            self._appeal_dialog.show()
+        from qtpy import QtModuleNotInstalledError
+        try:
+            if self._appeal_dialog is None:
+                self._appeal_dialog = InAppAppealDialog(self)
+
+            if not self._appeal_dialog.isVisible():
+                self._appeal_dialog.show()
+        except QtModuleNotInstalledError:
+            # QtWebEngineWidgets is optional
+            # So just open the URL in the default browser
+            # See https://github.com/spyder-ide/spyder/pull/23703
+            import webbrowser
+            webbrowser.open("https://opencollective.com/spyder")
+            pass
 
     # ---- StatusBarWidget API
     # -------------------------------------------------------------------------
